### PR TITLE
(#5490) include doc ID in conflict errors

### PR DESCRIPTION
**Pull request by alxndrsn**
Adds a property `suppliedDocId` to conflict error when trying to put or post an individual doc.

### CODE REVIEW

**Comment by alxndrsn** [comment]
I think it'd be an improvement to change both `err.suppliedDocId` here, and `err.requestedDocId` in https://github.com/pouchdb/pouchdb/issues/6529 to `err.docId`.

**Comment by daleharvey** [comment]
> I think it'd be an improvement to change both err.suppliedDocId here, 
> and err.requestedDocId in #6529 to err.docId.

I think this is a good idea, want to make this change? (apologies for the delay) . The tests look fine, the coverage failure looks like an intermittent and the code looks good, thanks :)

**Comment by alxndrsn** [comment]
👍 I'll update the property to `docId` in the next few days.

**Comment by alxndrsn** [comment]
@daleharvey relevant change for this PR made; change for 404 errors made at https://github.com/pouchdb/pouchdb/pull/6574

File: tests/integration/test.basics.js [hunk 1]
```diff
@@ -114,6 +114,46 @@ adapters.forEach(function (adapter) {
...
+      }).catch(function (err) {
+        err.should.have.property('status', 409);
+        err.docId.should.equal(savedDocId);
+      });
+    });
+
+    it('POSTed Conflicted doc should contain ID in error object', function () {
+      var db = new PouchDB(dbs.name);
+
+      console.log(db.post.toString());
```
Comment (daleharvey) [inline tests/integration/test.basics.js]: Need to remove the console :)

File: tests/integration/test.basics.js [hunk 2]
```diff
@@ -114,6 +114,46 @@ adapters.forEach(function (adapter) {
...
+
+      var savedDocId;
+
+      return db.post({}).then(function (info) {
+        console.log('saved', info);
```
Comment (daleharvey) [inline tests/integration/test.basics.js]: and here

File: tests/integration/test.basics.js [hunk 3]
```diff
@@ -114,6 +114,46 @@ adapters.forEach(function (adapter) {
       });
     });
 
+    it('PUTed Conflicted doc should contain ID in error object', function () {
+      var db = new PouchDB(dbs.name);
+
+      var savedDocId;
+
+      return db.post({}).then(function (info) {
+        console.log('saved', info);
```
Comment (daleharvey) [inline tests/integration/test.basics.js]: and here

**Pull request merged**